### DISCUSSION TITLE
Create Task

### DISCUSSION THREAD

**Discussion by UNiXMIT** [InteractionDesignFoundation/add-event-to-calendar-docs#40] [Q&A] [answered]
I am trying to discover if there is a URL I can use to create a task in Outlook.

How did you discover this if there is no official documentation? - https://github.com/InteractionDesignFoundation/add-event-to-calendar-docs/blob/main/services/outlook-web.md

If you can explain, perhaps I can try the same to find a URL to create a task.

Thanks for any advice.

**Comment by alies-dev** [comment, marked as answer]
@UNiXMIT 
I used reserve-engineering of js loaded on "Add event"  pages. Unfortunately, js filenames were pretty dynamic and I can't share any specific names

**Reply by UNiXMIT** [reply]
@lptn OK thanks. I'll dig in a little further and see what I can find.

**Comment by alies-dev** [comment]
Answering the second half, since it is the more useful one.

**Outlook.** No task deep link turned up. Outlook is the hard case: it does not parse the deep link in the browser at all. None of the parameter names (`rru`, `startdt`, `subject`, and so on) appear in the JavaScript chunks the compose page loads, and the query string disappears from the address bar once the form renders, so the mapping happens on the server. That means there is no table to read, and the only way to find a parameter is to guess a name and watch whether the form changes. Applied to tasks, nothing was found.

**Google** does have task related parameters, because its deep link parser is readable in the client bundle: `taskId`, `taskBundleId` and `ddl` (deadline) switch the editor into task mode instead of event mode. They reference an existing task rather than creating an arbitrary one, so they are not the "create a task from a URL" feature you are after either.

**The method**, in case you want to try it yourself: open the composer with parameters set, find the application bundle in the network panel, and search it for one parameter name you already know works. For Google and Yahoo the surrounding code contains the complete table of everything the client understands, which also proves the negatives. For Outlook, only black box replay works. There is a longer write-up of this in the sibling thread on how these were extracted.
